src/panels/config/devices/ha-config-devices-dashboard.ts

```diff
@@ -217,47 +212,48 @@ export class HaConfigDeviceDashboard extends LitElement {
         outputDevices = outputDevices.filter((device) => !device.disabled_by);
       }
 
-      outputDevices = outputDevices.map((device) => ({
-        ...device,
-        name: computeDeviceName(
-          device,
-          this.hass,
-          deviceEntityLookup[device.id]
-        ),
-        model:
-          device.model ||
-          `<${localize("ui.panel.config.devices.data_table.unknown")}>`,
-        manufacturer:
-          device.manufacturer ||
-          `<${localize("ui.panel.config.devices.data_table.unknown")}>`,
-        area:
-          device.area_id && areaLookup[device.area_id]
-            ? areaLookup[device.area_id].name
-            : "—",
-        integration: device.config_entries.length
-          ? device.config_entries
-              .filter((entId) => entId in entryLookup)
+      outputDevices = outputDevices.map((device) => {
+        const deviceEntries = entries.filter((entry) =>
+          device.config_entries.includes(entry.entry_id)
+        );
```

Comment:
The lookup was way quicker here, as we didn't have to filter all the entries for every device, but could loop over the entries of the device (which are a lot less)

Reply:
Sort the integrations after filtering? Then there are only a few integrations to sort?